docs/developers-manual.md

```diff
@@ -457,3 +457,57 @@ You can also populate the test data into the test environment using Docker:
 
 Make sure to refresh constants manually after doing this by logging in as admin, and clicking that
 option on the My Account page.
+
+### Spring Controller Testing
+
+#### Test data
+
+All test data is stored in `integration_test_data.sql`, so keep this in mind when writing the tests.
+
+#### Tests
+
+Spring controller tests should all extend `AbstractST`. The abstract class contains all the annotations needed to set up the test context/config/resources, as well as the mock user to run the tests. It also includes several template tests for common endpoints, for both the rest controllers and the model/view controllers.  The javadocs in `AbstractST.java` have more specific information on these.
+
+#### Running the tests
+Running the spring tests depends on whether you have the local database copy setup in a docker container or not.
+
+If you do, then you can run the tests as such:
+```
+mvn test-compile -DskipSTs=false -pl miso-web
+```
+To compile, then 
+```
+mvn -pl miso-web -Dmiso.it.mysql.url=jdbc:mysql://127.0.0.1:<port>/<test_db_name> -Dmiso.it.mysql.user=<user> -Dmiso.it.mysql.pw=<password> failsafe:integration-test -DskipSTs=false
+```
+ To run the tests.
+
+If not, then you can run the tests as follows
+```
+mvn clean verify -pl miso-web -DskipSTs=false
+```
```

Comment:
If you do have your local database set up in a Docker container, then you can use the following to compile and run the tests (replace the placeholders in the step to run the tests):
```
mvn test-compile -DskipSTs=false -pl miso-web

mvn -pl miso-web -Dmiso.it.mysql.url=jdbc:mysql://127.0.0.1:<port>/<test_db_name> -Dmiso.it.mysql.user=<user> -Dmiso.it.mysql.pw=<password> failsafe:integration-test -DskipSTs=false
```


If your local database is a MySQL install, use the following to run the tests:
```
mvn clean verify -pl miso-web -DskipSTs=false
```